tests/aws/services/secretsmanager/test_secretsmanager.py

```diff
@@ -2395,6 +2396,48 @@ def test_get_secret_value_errors(self, aws_client, create_secret, sm_snapshot):
             )
         sm_snapshot.match("mismatch_version_id_and_stage", exc.value.response)
 
+    @markers.aws.validated
+    @markers.snapshot.skip_snapshot_verify(paths=["$..CreatedDate"])
+    def test_get_secret_value(
+        self, aws_client, aws_http_client_factory, region_name, create_secret, sm_snapshot
+    ):
+        secret_name = short_uid()
+        secret_string = "footest"
+        secret_string_b64_encoded = base64.b64encode(secret_string.encode())
+
+        response = create_secret(
+            Name=secret_name,
+            SecretBinary=secret_string,
+        )
+
+        sm_snapshot.add_transformers_list(
+            sm_snapshot.transform.secretsmanager_secret_id_arn(response, 0)
+        )
+
+        secret_arn = response["ARN"]
+
+        secret_value_response = aws_client.secretsmanager.get_secret_value(SecretId=secret_arn)
+
+        sm_snapshot.match("secret_value_response", secret_value_response)
+
+        assert secret_value_response["SecretBinary"] == secret_string.encode()
+
+        client = aws_http_client_factory(
+            "secretsmanager", region=region_name, signer_factory=SigV4Auth
+        )
+        parameters = {"SecretId": secret_name}
+
+        headers = {
+            "X-Amz-Target": "secretsmanager.GetSecretValue",
+            "Content-Type": "application/x-amz-json-1.1",
+        }
```

Comment:
We should also add a test here to cover the previous case, basically the request that boto would usually send. This would make the particular difference between SDKs more obvious here.

Reply:
there are two snapshot assertions in this one is testing in boto, another one for http request.

Reply:
Sorry, I actually misunderstood the issue of the PR initially. It seems the Go SDK was quite a red herring actually in the initial report. All the clients are broken currently and return the wrong values when used with SecretBinary!